Assignments/Program.cs

```diff
@@ -1,3 +1,94 @@
-﻿using System;
+﻿// ---------------------------------------------------------------------------------------
+// Spark23 Assignments
+// Copyright (c) Metamation India.
+// ---------------------------------------------------------------------------------------
+// Program.cs
+// T26 - Implementing MyList<T>
+// ---------------------------------------------------------------------------------------
 
-Console.WriteLine("Hello, World!");
+using System;
+using System.Linq;
+
+namespace Spark23;
+
+public static class T26 {
+   public static void Main () {
+      MyList<int> list = new ();
+   }
+
+   /// <summary>Custom class of generic type list/> 
+   public class MyList<T> {
+      /// <summary>Returns the number of items present in the list</summary>
+      public int Count => mIndex;
+
+      /// <summary>Returns the capacity of the list</summary>
+      public int Capacity => mList.Length;
+
+      /// <summary>Property: Gets the value of specified index from the list 
+      ///  [or] Sets the value to the specified index of the list</summary>
+      public T this[int index] {
+         get {
+            if (index < 0 || index >= mIndex) 
+               throw new IndexOutOfRangeException ("Index was out of range. Must be non-negative and less than the size of the collection.");
+            return mList[index]; 
+         }
+         set {
+            if (index < 0 || index >= mIndex)
+               throw new IndexOutOfRangeException ("Index was out of range. Must be non-negative and less than the size of the collection.");
+            mList[index] = value;
+         }
+      }
+
+      /// <summary>Adds an given item to the end of the list</summary>
+      public void Add (T a) {
+         if (mIndex == mList.Length) Array.Resize (ref mList, mIndex * 2);
+         mList[mIndex++] = a;
+      }
+
+      /// <summary>Removes the given item from the list</summary>
+      public void Remove (T a) {
+         if (!mList.Contains (a)) throw new InvalidOperationException ("Value not found in the list");
+         int index = Array.FindIndex (mList, x => x.Equals(a));

```

Comment:
```suggestion
         int index = Array.FindIndex (mList, x => x.Equals (a));
```
better yet, don't use 'FindIndex' as it will take O(N) time. Step through the array till you find the element.